### PR TITLE
Fix drop_nulls cython threshold handling

### PR DESCRIPTION
**Pull request by harrism**
Corrects the default threshold in drop_nulls bindings to be the number of columns in the keys table (subset in python) if it exists, rather than the input table.

### CODE REVIEW

File: python/cudf/cudf/bindings/stream_compaction.pyx [hunk 1]
```diff
@@ -89,23 +100,44 @@ def apply_apply_boolean_mask(cols, mask):
...
+                                           else len(subset))
 
-    # Use threshold if specified, otherwise set it based on how
+    # Use `thresh` if specified, otherwise set threshold based on `how`
     if thresh:
-        keep_threshold = thresh
+        c_keep_threshold = thresh
     elif how == "all":
-        keep_threshold = 0
+        c_keep_threshold = 0
```
Comment (shwina) [inline python/cudf/cudf/bindings/stream_compaction.pyx]: Shouldn't this be `1` and not `0`. We need to keep the row if there's at least `1` non-null.

File: python/cudf/cudf/bindings/stream_compaction.pyx [hunk 2]
```diff
@@ -89,23 +100,44 @@ def apply_apply_boolean_mask(cols, mask):
...
 
-    cdef any_or_all drop_if
-    cdef gdf_size_type keep_threshold = len(cols)
+    # default: "any" means threshold should be number of key columns
+    cdef gdf_size_type c_keep_threshold = (len(cols) if subset is None
+                                           else len(subset))
 
-    # Use threshold if specified, otherwise set it based on how
+    # Use `thresh` if specified, otherwise set threshold based on `how`
     if thresh:
```
Comment (shwina) [inline python/cudf/cudf/bindings/stream_compaction.pyx]: This should be `if thresh is not None`.

`0` is considered `False` so if `thresh` is specified `0`, we fail to update `c_keep_threshold`.

Comment (harrism) [inline python/cudf/cudf/bindings/stream_compaction.pyx]:
```suggestion
    if thresh is not None:
```

File: python/cudf/cudf/bindings/stream_compaction.pyx (hunk 1 above)
Comment (harrism) [inline python/cudf/cudf/bindings/stream_compaction.pyx]:
```suggestion
        c_keep_threshold = 1
```

Comment (harrism) [reply python/cudf/cudf/bindings/stream_compaction.pyx]: Fixed, thanks.

File: python/cudf/cudf/bindings/stream_compaction.pyx (hunk 2 above)
Comment (harrism) [reply python/cudf/cudf/bindings/stream_compaction.pyx]: Right, None quacks like a False.

File: python/cudf/cudf/bindings/stream_compaction.pyx [hunk 3]
```diff
@@ -89,23 +100,44 @@ def apply_apply_boolean_mask(cols, mask):
 
 
 def apply_drop_nulls(cols, how="any", subset=None, thresh=None):
+    """
+    Drops null rows from cols.
```
Comment (harrism) [inline python/cudf/cudf/bindings/stream_compaction.pyx]:
```suggestion
    Drops null rows from cols.
```

**Pull request merged**